src/linter/engine.mjs

```diff
@@ -1,29 +1,29 @@
 'use strict';
 
 /**
- * Creates a linter engine instance to validate ApiDocMetadataEntry entries
+ * Creates a linter engine instance to validate mdast trees.
  *
  * @param {import('./types').LintRule[]} rules Lint rules to validate the entries against
  */
 const createLinterEngine = rules => {
   /**
-   * Validates an array of ApiDocMetadataEntry entries against all defined rules
+   * Validates an array of mdast trees against all defined rules
    *
-   * @param {ApiDocMetadataEntry[]} entries
+   * @param {import('mdast').Root[]} ast
    * @returns {import('./types').LintIssue[]}
    */
-  const lintAll = entries => {
+  const lint = ast => {
     const issues = [];
 
     for (const rule of rules) {
-      issues.push(...rule(entries));
+      issues.push(...rule(ast));
```

Comment:
WDYT about there being be a way for the linter to accept raw OR ast, depending on what it needs. For some scenarios, getting the AST may be preferable

Reply:
I don't have a strong opinion about this, I can make a simple test so we can decide what is the best option.

Reply:
Giving another look, and I think it's fine like this. Looks really good :-)